From 831342f5033d2a5188329c1e4a8306b5278b83c8 Mon Sep 17 00:00:00 2001
From: Tony Mak <tonymak@google.com>
Date: Mon, 26 Mar 2018 12:38:04 +0100
Subject: [PATCH 171/306] Get application info in the correct user id

Fixes: 76396560

Test: Create work profile and launch hangouts in work profile.
Change-Id: Ic3727ac899153b6a96dd9b6d6921c32ba0bbd93e
(cherry picked from commit 352dc0b443a084d29d00f3b31b67917435afe1a3)
---
 .../com/android/server/connectivity/PermissionMonitor.java    | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/connectivity/PermissionMonitor.java b/services/core/java/com/android/server/connectivity/PermissionMonitor.java
index d578e959314..e471c7d84b5 100644
--- a/services/core/java/com/android/server/connectivity/PermissionMonitor.java
+++ b/services/core/java/com/android/server/connectivity/PermissionMonitor.java
@@ -202,7 +202,9 @@ public class PermissionMonitor {
             // Only using the first package name. There may be multiple names if multiple
             // apps share the same UID, but in that case they also share permissions so
             // querying with any of the names will return the same results.
-            final PackageInfo app = mPackageManager.getPackageInfo(names[0], GET_PERMISSIONS);
+            int userId = UserHandle.getUserId(uid);
+            final PackageInfo app = mPackageManager.getPackageInfoAsUser(
+                    names[0], GET_PERMISSIONS, userId);
             return hasUseBackgroundNetworksPermission(app);
         } catch (NameNotFoundException e) {
             // App not found.
-- 
2.17.1

